app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class InvalidCommandFormat(Exception):
+    pass
+
+
+class InvalidCommand(Exception):
+    pass
+
+
+class SameFileError(Exception):
+    pass
+
+
+class SourceFileNotFoundError(Exception):
+    pass
+
+
+class PermissionDeniedError(Exception):
+    pass
+
+
+def copy_file(command: str) -> None:
+    commands = command.split()
+
+    if len(commands) != 3:
+        raise InvalidCommandFormat("Invalid command format. "
+                                   "Usage: cp source_file destination_file")
+
+    command_name, source_file, destination_file = commands
+
+    if command_name != "cp":
+        raise InvalidCommand("Invalid command. Use 'cp' to copy files.")
+
+    if source_file == destination_file:
+        raise SameFileError("Source file and destination file are the same.")
+
+    try:
+        with (open(source_file, "r") as file_in,
```

Comment:
This is the only piece of code where you really need exceptions. OSError should cover all cases if you try to copy an invalid / non-existing file.
Also you can simplify this by using `shutil` library